File: Modules/CustomerManagement/Customer/CustomerForm.cs
namespace MapPin.CustomerManagement.Forms;

[FormScript("CustomerManagement.Customer")]
[BasedOnRow(typeof(CustomerRow), CheckNames = true)]
public class CustomerForm
{
    [HalfWidth]
    public string CustomerName { get; set; }

    [HalfWidth]
    public string ContactName { get; set; }

    [HalfWidth]
    public string Email { get; set; }

    [HalfWidth]
    public string Phone { get; set; }

    public string Address { get; set; }

    [HalfWidth, LookupEditor("CustomerManagement.Il")]
    public string City { get; set; }

    [HalfWidth, LookupEditor("CustomerManagement.Ilce", CascadeFrom = "City", CascadeField = "IlId")]
    public string District { get; set; }

    //[NoteEditor]
    public string Notes { get; set; }
}

File: Modules/ServerTypes/CustomerManagement/CustomerForm.ts
import { initFormType, LookupEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface CustomerForm {
    CustomerName: StringEditor;
    ContactName: StringEditor;
    Email: StringEditor;
    Phone: StringEditor;
    Address: StringEditor;
    City: LookupEditor;
    District: LookupEditor;
    Notes: StringEditor;
}

export class CustomerForm extends PrefixedContext {
    static readonly formKey = 'CustomerManagement.Customer';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CustomerForm.init) {
            CustomerForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;

            initFormType(CustomerForm, [
                'CustomerName', w0,
                'ContactName', w0,
                'Email', w0,
                'Phone', w0,
                'Address', w0,
                'City', w1,
                'District', w1,
                'Notes', w0
            ]);
        }
    }
}